Assets/Scripts/MovingPlatform.cs

```diff
@@ -0,0 +1,32 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class MovingPlatform : MonoBehaviour
+{
+    public List<Transform> waypoints;
+    public float moveSpeed;
+    public int target;
+
+    void Update()
+    {
+        //receive data
+        transform.position = Vector3.MoveTowards(transform.position, waypoints[target].position, moveSpeed * Time.deltaTime);
```

Comment:
Replace with rigidbody.MovePosition(Vector3.MoveTowards(transform.position, waypoints[target].position, moveSpeed * Time.deltaTime)) and put all of this in FixedUpdate instead.